src/java.desktop/macosx/classes/sun/awt/CGraphicsDevice.java

```diff
@@ -65,9 +68,20 @@
 
     public CGraphicsDevice(final int displayID) {
         this.displayID = displayID;
-        config = MacOSFlags.isMetalEnabled() ?
-                MTLGraphicsConfig.getConfig(this, displayID, 0) :
-                CGLGraphicsConfig.getConfig(this);
+
+        if (MacOSFlags.isMetalEnabled()) {
```

Comment:
I guess this flag really means "requested". It does not mean "supported and enabled".
And we also have a boolean called isMetalEnabled which is closer to the truth.